### PR TITLE
Backend monitor and overview

### LINKED ISSUE #994: Adding a backend_overview
I think it is important to be able to view the backend status. I find myself writing this code lots 

```python
for backend in [IBMQ.backends(), Aer.backends()]:
     print(backend.status())
```
and even this is not enough. I would like to be able to have a terminal or a section of my Juypiter (or even app) telling me continually the backend status. 

I would like to write something like 

``` python
backend_monitor([IBMQ, Aer,...])
``` 

and this makes an output for me like 

```
{'pending_jobs': 10, 'name': 'ibmqx4', 'operational': True}
{'pending_jobs': 8, 'name': 'ibmqx5', 'operational': False}
{'pending_jobs': 0, 'name': 'ibmqx2', 'operational': False}
{'pending_jobs': 54, 'name': 'ibmq_16_melbourne', 'operational': True}
{'pending_jobs': 0, 'name': 'ibmq_qasm_simulator', 'operational': True}
{'name': 'qasm_simulator', 'operational': True, 'pending_jobs': 0}
{'name': 'qasm_simulator_py', 'operational': True, 'pending_jobs': 0}
{'name': 'statevector_simulator', 'operational': True, 'pending_jobs': 0}
{'name': 'statevector_simulator_py', 'operational': True, 'pending_jobs': 0}
{'name': 'unitary_simulator', 'operational': True, 'pending_jobs': 0}
{'name': 'clifford_simulator', 'operational': True, 'pending_jobs': 0}
```
I think we need to add 'provider': 'IBMQ' (also add this to the spec) and "status_msg": "This is a status message" where the status_msg is set by the backend controller (down for maintence, retireded)...

I think this should continue until I quit it, but this we can discuss more. Like the job_monitor it should live in wrapper or tools. I still need @delapuente to convince me what the difference is

### PR DESCRIPTION
**Pull request by nonhermitian**
Fixes #994 
### Summary
These are Jupyter magics that gives the overview of all the backends that have been loaded using `IBMQ.load_accounts()` (`backend_overview`) or a detailed view of a single backend (`backend_monitor`).  The `backend_overview` updates automatically, while the monitor is currently static.  Additionally, there is a `plot_coupling_map` function in the `visualization` module.

# Overview
overview

# Monitor
monitor1
monitor2
monitor3
monitor4
monitor5

### Details and comments
Things to do:

- [x] The coupling map drawer needs a better way of handling the required equal aspect ratio.
- [x] The figure size used in the monitor is hacked (backend specific) because of the above issue.
- [x] The coupling map drawer should probably be moved elsewhere as its usefulness goes beyond this PR.
- [x] Need to verify that the upper bound on the pending jobs correctly extends itself if the number of jobs goes above the default 50.
- [x] The background function `update_backend_info`  should gracefully handle a status check not working, and should probably terminate if none of the backends are responding (i.e there is no network connection to anything).(Each backend_widget has a is_alive attr, added just for this purpose)
- [x] Allow for changing the status check interval via a magic arg.
- [ ] The `mpl_data` is the physical layout of a device on a grid.  This should be something that the backend returns.
- [x] Make the damn linter happy.

### CODE REVIEW

**Comment by nonhermitian** [comment]
@mtreinish, @jaygambetta this is now ready for review.  I am punting the move of the coupling_map plotter to later, as we need to have the backends return the info that mpl needs (this is in progress).

**Comment by jaygambetta** [comment]
@nonhermitian as we discussed i think this should have some small changes
- backend_overview
-a function backend_overview which is blocking that returns the backend end overview

I would like the code to look similar in organization to the job monitor. 

Also lets link this to issue #994 and #1010

**Comment by jaygambetta** [comment]
@nonhermitian did you get the backend_overview function

File: qiskit/tools/jupyter/__init__.py [hunk 1]
```diff
@@ -10,9 +10,11 @@
 
 from IPython import get_ipython          # pylint: disable=import-error
 from .jupyter_magics import (ProgressBarMagic, StatusMagic)
-from .progressbar import HTMLProgressBar, TextProgressBar
+from .progressbar import HTMLProgressBar
+from .backend_monitor import BackendMonitor
```
Comment (jaygambetta) [inline qiskit/tools/jupyter/__init__.py]: change name to backend_overview

**Comment by nonhermitian** [comment]
I am working on getting both backend overview and monitors (text and notebook) in this pr.

**Comment by nonhermitian** [comment]
@jaygambetta, this is ready to go.

**Comment by nonhermitian** [comment]
This is on hold until the backends are spec compliant.

**Comment by jaygambetta** [comment]
man @nonhermitian were you watching this in real time.

**Pull request merged**